### PR TITLE
feat(accordion): support div as title

### PR DESCRIPTION
**Pull request by asudoh**
Refs IBM/carbon-components-react#1178.

#### Changelog

**New**

- Some style to support `<div>` as accordion title.

#### Testing / Reviewing

Testing should make sure accordion is not broken, even if a consumer keeps using `<p>` as accordion title.

### CODE REVIEW

**Review by joshblack — APPROVED** [review]

File: src/components/accordion/_accordion.scss [hunk 1]
```diff
@@ -71,6 +71,9 @@
 
   .#{$prefix}--accordion__title {
     margin: $accordion-title-margin;
+    font-size: 1rem;
```
Comment (joshblack) [inline src/components/accordion/_accordion.scss]: Is there an easy way to keep this in sync with the `p` styles?

**Review by alisonjoseph — APPROVED** [review]

File: src/components/accordion/_accordion.scss (hunk 1 above)
Comment (asudoh) [reply src/components/accordion/_accordion.scss]: Mix-in is something that comes to mind, but it raises a question of if we want to include all the styles of `<p>` or want to maintain a subset. Thoughts?

Comment (alisonjoseph) [reply src/components/accordion/_accordion.scss]: I'm not sure why this would need to stay in sync with `p` styles?

Comment (asudoh) [reply src/components/accordion/_accordion.scss]: Good question @alisonjoseph! I think the talk was originally from the fact that this PR is creating an accordion header style that works both for `<p>` and `<div>`, and I copied a subset of `<p>` style here. Something that comes to my mind in this topic is https://twitter.com/marisafari/status/923571338765148161, but having brought it up, I think you made a good point that we just need to define a style that are not affected by future `<p>`/`<div>` style changes?

Comment (joshblack) [reply src/components/accordion/_accordion.scss]: Yeah, +1 to what @asudoh said. The node originally was a `p` tag and so it inherited `p` styles. Just want to make sure the type styles stay in sync with whatever they need to be, whether that's `p` or using our `typescale` mixin. Either way!

Comment (asudoh) [reply src/components/accordion/_accordion.scss]: 👍 for `@include typescale()` - Made the change, thank you for the suggestion!

**Pull request merged**